frontend/src/screens/analysis/analysis-tutorial.tsx

```diff
@@ -0,0 +1,165 @@
+import { Box, Modal, Row, Col, Button, ResearcherButton } from '@components';
+import { React, styled, useState } from '@common';
+import { useApi } from '@lib';
+import { noop } from 'lodash-es';
+import Tutorial1 from '@images/analysis/tutorial1.svg'
+import Tutorial2 from '@images/analysis/tutorial2.svg'
+import Tutorial3 from '@images/analysis/tutorial3.svg'
+import { colors } from '@theme';
+
+export const AnalysisTutorial: FC<{show: boolean}> = ({ show }) => {
+    const api = useApi()
+    const [showTutorial, setShowTutorial] = useState(show)
+    const [step, setStep] = useState(0)
+
+    const closeTutorial = () => {
+        api.updatePreferences({ updatePreferences: { preferences: { hasViewedAnalysisTutorial: true } } }).then(() => {
+            setShowTutorial(false)
+        })
+    }
+
+    const renderStep = (step: number, setStep: (step: number) => void) => {
+        switch(step) {
+            case 0:
+                return <Step1 step={step} setStep={setStep} />
+            case 1:
+                return <Step2 step={step} setStep={setStep} />
+            case 2:
+                return <Step3 step={step} setStep={setStep} close={closeTutorial} />
+        }
+    }
```

Comment:
This doesn't appear to rely on any closure variables, so why not move it out of the AnalysisTutorial function and make it a "real" component `TutorialStep` or something, then you can render it below as `<TutorialStep index={step} onChange={stetStep} />`